rr_iarrc/src/finish_line_watcher/finish_line_watcher.cpp

```diff
@@ -78,23 +84,122 @@ void ImageCB(const sensor_msgs::ImageConstPtr& msg) {
     frame = cv_ptr->image;
     blockEnvironment(frame);
 
-    auto count = countNonZero(frame);
+    // Matrix used for detecting contours
+    Mat contourImg;
+    // Matrix used to draw debug info
+    Mat debugDrawing = Mat::zeros(frame.size(), CV_8UC3);
+    cv::cvtColor(frame, debugDrawing, cv::COLOR_GRAY2BGR);
+
+    // Find contours
+    vector<vector<Point> > contours;
+    vector<Vec4i> hierarchy;
+    cv::Canny(frame, contourImg, canny_thresh1, canny_thresh2, 3);
```

Comment:
Still confused why you did Canny here instead of going straight to findContours.

Reply:
[here](https://docs.opencv.org/2.4/doc/tutorials/imgproc/shapedescriptors/find_contours/find_contours.html) , but it isnt needed and removed now and works the same

Reply:
You already have a binary image so you don't need it